### PR TITLE
New C++ API: Add unit tests for setInfo, setLogic, setOption.

### PR DESCRIPTION
**Pull request by aniemetz**

### CODE REVIEW

File: src/api/cvc4cpp.cpp [hunk 1]
```diff
@@ -3554,36 +3555,87 @@ void Solver::reset(void) const { d_smtEngine->reset(); }
...
-  // if keyword == "cvc4-logic": value must be string
-  // if keyword == "status": must be sat, unsat or unknown
-  // if keyword == "smt-lib-version": supported?
+  bool is_cvc4_keyword = false;
+
+  /* Check for CVC4-specific info keys (prefixed with "cvc4-" or "cvc4_") */
+  if (keyword.length() > 5)
+  {
+    std::string prefix = keyword.substr(0, 5);
+    if (prefix == "cvc4-" || prefix == "cvc4_")
```
Comment (4tXJ7f) [inline src/api/cvc4cpp.cpp]: o_O

That's a weird feature, we should maybe get rid of it in the future...

File: src/smt/smt_engine.h [hunk 2]
```diff
@@ -456,6 +456,14 @@ class CVC4_PUBLIC SmtEngine {
    */
   ~SmtEngine();
 
+  /**
+   * Return true if this SmtEngine is fully initialized (post-construction).
+   * This post-construction initialization is automatically triggered by the
+   * use of the SmtEngine; e.g. when setLogic() is called, or the first
```
[the remark this answers is not in the document]
Comment (aniemetz) [reply src/smt/smt_engine.h]: Hm, interesting. Then we should fix the comment in SmtEngine, too. That's were this is from. Will do.

File: src/api/cvc4cpp.cpp (hunk 1 above)
Comment (aniemetz) [reply src/api/cvc4cpp.cpp]: Haha, yes, my thoughts. @ajreynol can you maybe chip in?

File: src/smt/smt_engine.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (4tXJ7f) [reply src/smt/smt_engine.h]: Yeah, you are right... Thanks for the fixing that.

**Pull request merged**